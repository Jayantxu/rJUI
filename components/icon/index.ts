import Icon from './icon';
import Pasteicon from './pasteIcon';
export { Pasteicon };
export default Icon;