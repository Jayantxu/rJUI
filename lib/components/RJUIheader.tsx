import React from 'react';
import Head from 'next/head';

interface PAGETITLE {
    title: string
}
const pageHeader: React.FC<{pageTitle: PAGETITLE}> = ({ pageTitle }) => (
    <Head>
        <title>{pageTitle.title ? `${pageTitle.title} | ` : ''}RJUI - Jayant React UI</title>
    </Head>
);
export default pageHeader;