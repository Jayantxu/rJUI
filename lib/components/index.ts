export { default as Playground } from './playground';
export { default as RJUILayout } from './RJUILayer';