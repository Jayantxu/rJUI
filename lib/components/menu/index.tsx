import React from 'react';
import RJUILink from './RJUILink';
import RJUISticker from './RJUISticker';

const Menu: React.FC<unknown> = () => {
    return (
    <div className="rjui-header">
        <RJUILink />
        <RJUISticker />
        <style global jsx>{`
        `}</style>
    </div>
  );
};

export default Menu;
