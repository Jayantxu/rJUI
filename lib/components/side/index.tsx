import React, { useMemo } from 'react';
import Router, { useRouter } from 'next/router';
import { menuHeadDatas, MenuArrInfo } from '../../data/menuJson';

const MenuAside: React.FC<unknown> = () => {
    const { pathname } = useRouter();

    const menuInfo = useMemo(() => {
        const tabName = pathname.split('/').filter(item => !!item)[0];
        let menuArr: MenuArrInfo[] = [];
        const notChildren: MenuArrInfo[] = [];
        menuArr = menuHeadDatas.filter(item => { 
            return item.name === tabName;
        });
        if(!menuArr.length) {
            return notChildren;
        }
        return menuArr[0].children;
    }, [pathname]);

    const menuUnitClick = (menu: MenuArrInfo) => {
        if(!menu.url) return;
        Router.push(menu.url);
    };
    const judgeActiveMenu = (menuUrl: string): boolean => {
        if(pathname === menuUrl) {
            return true;
        }
        return false;
    };
    return (
        <>
            <div className="rjui-aside-menu">
                {menuInfo.map((menuUnit, midx) => (
                    <div className="rjui-aside-menu-unit" key={menuUnit.name + midx}>
                        <span className="rjui-menu-unit-title">{menuUnit.name}</span>
                        <div className="rjui-menu-unit-content">
                            {menuUnit.children.map((menu, mjdx) => (
                                <div onClick={() => {menuUnitClick(menu);}} key={menu.menuName + mjdx} 
                                    className={`rjui-menu-unit rjui-hover ${judgeActiveMenu(menu.url) ? 'is-active-menu': '' } `}>
                                    {menu.menuName}
                                </div>
                            ))}
                        </div>
                    </div>
                ))}
            </div>
            <style jsx>{`
                .rjui-aside-menu {
                    display: flex;
                    flex-direction: column;
                    height: 100%;
                }
                .rjui-aside-menu-unit {
                    margin-bottom: 1rem;
                }
                .rjui-menu-unit-title {
                    font-size: 0.875rem;
                    transition: all 0.2s ease 0s;
                    color: #00000073;
                    letter-spacing: 1.3px;
                }
                .rjui-menu-unit-content {
                    margin-top: 0.4rem;
                }
                .rjui-menu-unit {
                    height: 2.2rem;
                    line-height: 2.2rem;
                    padding-left: 0.3rem;
                    font-size: 0.875rem;
                    color: #314659;
                }
                .rjui-menu-unit:hover {
                    color: #1890ff;
                }
                .is-active-menu {
                    color: #1890ff;
                }
            `}</style>
        </>
    );
};
export default MenuAside;