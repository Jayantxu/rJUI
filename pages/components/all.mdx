import RJUILayout from 'lib/components/RJUILayer';

export const pageTitle = {
    title: "All MDX",
};

## all.mdx

THIS IS all.mdx

## all.mdx

THIS IS all.mdx

## all.mdx

THIS IS all.mdx

## all.mdx

THIS IS all.mdx

## all.mdx

THIS IS all.mdx

## all.mdx

THIS IS all.mdx

## all.mdx

THIS IS all.mdx

## all.mdx

THIS IS all.mdx

## all.mdx

THIS IS all.mdx
## all.mdx

THIS IS all.mdx
## all.mdx

THIS IS all.mdx
## all.mdx

THIS IS all.mdx
## all.mdx

THIS IS all.mdx
## all.mdx

THIS IS all.mdx

export default ({ children }) => <RJUILayout pageTitle={pageTitle}>{children}</RJUILayout>
