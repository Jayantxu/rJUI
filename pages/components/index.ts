import redirect from 'lib/redirect';

export default redirect('/components/all');