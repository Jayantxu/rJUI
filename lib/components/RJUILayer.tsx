// # RJUIMainLayer
import { ReactNode } from 'react';
import RJUIHeader from './RJUIheader';
import { useConfigs } from 'lib/data/context';
import MenuAside from './side/index';
import { MobileMenuAside } from './side/RJUIMobileAside';

interface PAGETITLE {
    title: string
}
interface LayerMeta {
    children: ReactNode,
    pageTitle: PAGETITLE
}

export const RJUILayer: React.FC<LayerMeta> = ({ children, pageTitle }: LayerMeta) => {
    const { headerTabFixed, isMobileType } = useConfigs();
    return (
        <>
            <section className="rjui-layer">
                <RJUIHeader pageTitle={pageTitle} />
                <aside className={`${isMobileType ? 'rjui-side' : '' }`}>
                    { isMobileType ? <MenuAside /> : <MobileMenuAside /> }
                </aside>
                { isMobileType && <div className="rjui-i-main-distance"></div>}
                <div className="rjui-main-content">{ children }</div>
            </section>
            <style jsx>{`
                .rjui-layer {
                    min-height: calc(100vh - 60px - 48px);
                    max-width: 70rem;
                    margin: 0 auto;
                    padding: 0 1rem;
                    display: flex;
                    box-sizing: border-box;
                    position: relative;
                }
                .rjui-side {
                    width: 200px;
                    height: calc(100% - 8rem);
                    margin-right: 20px;
                    flex-shrink: 0;
                    -webkit-overflow-scrolling: touch;
                    -webkit-flex-shrink: 0;
                    bottom: 2rem;
                    position: fixed;
                    top: calc(60px + 48px + ${headerTabFixed ? '-40px' : '10px'});
                    transition: top 0.25s;
                    z-index: 100;
                    overflow: auto;
                }
                .rjui-side::-webkit-scrollbar {
                    width: 0;
                    background-color: transparent;
                }
                .rjui-i-main-distance {
                    width: 220px;
                    flex-shrink: 0;
                    height: 100vh;
                }
                .rjui-main-content {
                    flex-grow: 1;
                }
            `}</style>
        </>
    );
};

export default RJUILayer;