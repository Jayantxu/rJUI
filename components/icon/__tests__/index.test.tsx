import React from 'react';
import { mount, render, shallow } from 'enzyme';
import { Icon } from 'components';

describe('Icon', () => {
    it('正确渲染Icon', () => {
        const icon = shallow(<Icon name="rJUI"/>);
        expect(() => icon.unmount()).not.toThrow();
    });
    
    it('Icon快照', () => {
        const icon = render(<Icon />);
        expect(icon).toMatchSnapshot();
    });
    
    it('正确渲染相关类名', () => {
        const icon = render(<Icon name="rJUI" className="myTest icon-red"/>);
        expect(icon.hasClass('myTest icon-red')).toBe(true);
        expect(icon.hasClass('otherClassName')).toBe(false);
    });

    it('正确挂载不同尺寸', () => {
        const iconWrap = mount(
            <div>
                <Icon name="rJUI" size="10" />
                <Icon name="rJUI" size="20" />
                <Icon name="rJUI" size="30" />
            </div>
        );
        // expect(iconWrap.childAt(1)).toBe(true);
        expect(() => iconWrap.unmount()).not.toThrow();
    });
    
    //   it('should stacked when avatars are in a group', () => {
    //     const group = render(
    //       <Avatar.Group>
    //         <Avatar />
    //         <Avatar />
    //       </Avatar.Group>,
    //     )
    //     expect(group).toMatchSnapshot()
    //   })
    
    //   it('should show count in group', () => {
    //     const count = 20
    //     const group = shallow(<Avatar.Group count={count} />)
    //     const text = group.find('.count').text()
    //     expect(text).toContain(count)
    //   })
});