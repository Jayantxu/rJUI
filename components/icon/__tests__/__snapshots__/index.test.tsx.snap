// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Icon快照 1`] = `
initialize {
  "0": Node {
    "attribs": Object {
      "aria-label": "",
      "class": "jsx-3186953505 rjui-icon  ",
      "role": "img",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "class": "jsx-3186953505",
          "fill": "currentColor",
          "height": "1em",
          "width": "1em",
        },
        "children": Array [
          Node {
            "attribs": Object {
              "class": "jsx-3186953505",
              "href": "#",
            },
            "children": Array [],
            "name": "use",
            "namespace": "http://www.w3.org/2000/svg",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "href": "http://www.w3.org/1999/xlink",
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "href": "xlink",
            },
          },
        ],
        "name": "svg",
        "namespace": "http://www.w3.org/2000/svg",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "fill": undefined,
          "height": undefined,
          "width": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "fill": undefined,
          "height": undefined,
          "width": undefined,
        },
      },
    ],
    "name": "span",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "aria-label": undefined,
      "class": undefined,
      "role": undefined,
    },
    "x-attribsPrefix": Object {
      "aria-label": undefined,
      "class": undefined,
      "role": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
